exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -236,25 +238,51 @@ func (prwe *prwExporter) execute(ctx context.Context, writeReq *prompb.WriteRequ
 	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
 	req.Header.Set("User-Agent", prwe.userAgentHeader)
 
-	resp, err := prwe.client.Do(req)
-	if err != nil {
+	// Retry function
+	retryFn := func() error {
```

Comment:
why are you still implementing your own retry logic?  retrying on specific status codes only is an http convention. 

If you use this library, you will not need to do anything, just instantiate the client:

https://github.com/hashicorp/go-retryablehttp#getting-a-stdlib-httpclient-with-retries

you just have to replace the internal client in the struct returned by `retryablehttp.NewClient()` by the client returned from `prwe.clientSettings.ToClient(host, prwe.settings)`.

Reply:
reference: https://github.com/hashicorp/go-retryablehttp/issues/119